### ISSUE TITLE
Context menu with menu key broken

### ISSUE THREAD

**Issue by vendelin8** [mate-desktop/caja#1198]
#### Expected behaviour
Show context menu.

#### Actual behaviour
Nothing.

#### Steps to reproduce the behaviour
Press menu key on a selected item. It's only for the menu key, mouse right click shows the context menu. Menu key works otherwise, eg. in the browser.

#### MATE general version
1.22.0-1

#### Package version
1.22.0-1

#### Linux Distribution
Arch Linux x64

#### Link to downstream report of your Distribution
I'm not sure what to write here, but what about https://www.archlinux.org/packages/community/x86_64/caja/ ? :D

**Comment by lukefromdc** [comment]
Has this ever worked to your knowledge? This is the first I've heard of it (my default workflow is mouse-driven so I have to be told what to test on keybindings) so I have never seen this before. I just tested this with the text view in Pluma and it  works fine there, but I'm trying to figure out whether this used to work in Caja's and is now broken, or was simply never added.

It DOES work with a treeview (list view) in a Caja navigation window, but not with an icon view or a compact view. That might be a feature from GTK's treeview itself, wheras the icon view is a custom widget.

**Comment by vendelin8** [comment]
I'm using Mate for more than 5 years, and yes, it was very weird, that I clicked it yesterday, and it didn't work. So I'm pretty sure, that it was working before (on 1.20 and since I can remember).

Yes, it works in the Caja side navigation bar, and in Pluma too.

I'm using folder-color-switcher, I uninstalled it just to check if it changes this behaviour, but it's still not working.

**Comment by vendelin8** [comment]
And yes, I just reverted to 1.20.3-2, and it works there. Please let me know if I can help in any way.

**Comment by lukefromdc** [comment]
OK, that means something must have changed in the icon view after 1.20.3, so we are looking for a commit between 1.20.3 and 1.22.0 that broke this

**Comment by lukefromdc** [comment]
OK, when running Caja from terminal I instantly get 

`(caja:8752): Gtk-CRITICAL **: 02:45:59.033: gtk_menu_popup_at_rect: assertion 'GDK_IS_WINDOW (rect_window)' failed
`
when striking the menu key while the mouse is over a caja icon view.  That means the offending commit was probably a replacement of deprecated gtk_menu-popup and now I know where to look

**Comment by lukefromdc** [comment]
Try 
https://github.com/mate-desktop/caja/pull/1200
if you can compile caja locally

**Comment by sc0w** [comment]
@lukefromdc 

culprit commit: https://github.com/mate-desktop/caja/commit/c0f159de3fc1fdce082acef36044016e3f7b3527 merged in PR https://github.com/mate-desktop/caja/issues/996

**Comment by vendelin8** [comment]
> @lukefromdc
> 
> culprit commit: [c0f159d](https://github.com/mate-desktop/caja/commit/c0f159de3fc1fdce082acef36044016e3f7b3527) merged in PR #997 (fix for #996)

Yes, that's what I've changed back, and it works fine again. I suppose this is something you'd like to have because of the other issue. So what would be a real solution for this?

**Comment by lukefromdc** [comment]
Note that in 
https://github.com/mate-desktop/caja/pull/997 I changed the entire `eel_pop_up_context_menu` function in eel/eel-gtk-extensions.c while in https://github.com/mate-desktop/caja/pull/1200 I pass NULL as event to that function only when it is called from `icon_container_context_click_background_callback` in  fm-icon-view.c , so right-click context menus on the pathbar work fine with this.
In other words, the only time we want to pass NULL(meaning use the currrent GdkEvent) is when we are calling the icon view's context menu, so pass NULL in that case and pass the button event in all the others, with  `eel_pop_up_context_menu`  then passing whatever we sent it to `gtk_menu_popup_at_pointer` thus we can have it both ways with this.

**Comment by vendelin8** [comment]
This works fine for me. Thanks!

One more question: how do you open Caja from the terminal? If I type `caja`, it logs me out.

**Comment by lukefromdc** [comment]
Invoking caja from the command line normally connects to the existing instance of caja and opens a new window. To test this and see error messages from a new caja instance, I used "sudo caja" as a different user (root) gets a new instance of caja instead of a new window in the same process.

If you are being logged out when invoking caja from the command line, you  may have just found a bug in mate-session-manager. Only running "caja --quit" or "killall caja" from terminal should terminate the running instance of caja, and then caja should immediately restart, with the desktop icons thus disappearing and quickly reappearing. Your session should continue.

**Comment by vendelin8** [comment]
Ok, "sudo caja" works fine. I restarted my computer since the last time, and now it doesn't log me out. I guess I made too many changes. Thanks.

I think this can be closed. Or do you want to wait until the next release?

**Comment by lukefromdc** [comment]
We can close this(it should happen automatically) when the PR is merged to
master.
